Sources/Datadog/URLSessionAutoInstrumentation/Interception/URLFiltering/FirstPartyURLsFilter.swift

```diff
@@ -12,13 +12,10 @@ internal struct FirstPartyURLsFilter {
     /// "example.com", "api.example.com", but not "foo.com".
     private let regex: String
 
-    init(configuration: FeaturesConfiguration.URLSessionAutoInstrumentation) {
-        // pattern = "^(.*\\.)*tracedHost1|^(.*\\.)*tracedHost2|..."
-        self.regex = configuration.userDefinedFirstPartyHosts.map { host in
-            let escaped = NSRegularExpression.escapedPattern(for: host)
-            return "^(.*\\.)*\(escaped)$"
-        }
-        .joined(separator: "|")
+    init(hosts: Set<String>) {
+        // pattern = "^(.*\\.)*tracedHost1|tracedHost2|...$"
+        let escapedHosts = hosts.map { NSRegularExpression.escapedPattern(for: $0) }.joined(separator: "|")
+        self.regex = "^(.*\\.)*\(escapedHosts)$"
```

Comment:
Maybe this is anecdotal, but I've seen that previous regex would take 1.something to 20x more time than this slightly different regex, while the unit tests still pass.

Reply:
super! @xgouchet @mariusc83 you may want to change your regex too